src/main/java/unicash/model/commons/Amount.java

```diff
@@ -3,23 +3,65 @@
 import static unicash.commons.util.AppUtil.checkArgument;
 
 /**
- * Represents a Transaction's amount.
+ * Represents a Transaction's amount. Amounts are stored as a double with
+ * a maximum precision of 2 decimal places as with most general financial
+ * transactions in the real world. All amounts must be positive, and amounts
+ * exceeding the above mandated precision will be rounded (not truncated) to
+ * 2 decimal places. Amounts, if represented as a string, will be prefixed with
+ * a currency indicator.
  */
 public class Amount {
```

Comment:
Was wondering, but why is duplicate work to add constraints done in this PR? I believe #97 is also handling the constraints, is there a reason why we duplicated the constraint handling? 

Only asking since UI modifications does not seem to entail including constraint checking. 

No issues here but maybe we can keep PRs to be local to the problems we're addressing and avoid adding changes outside of the scope to avoid potential merge conflicts/duplicate work done, this way it's also easier to know at a glance what the PRs are addressing

Reply:
Yes actually this PR was not supposed to deal with the Amount class at all, but due to some formatting discrepanices of Amounts (also mentioned in [this](https://github.com/AY2324S1-CS2103-T16-3/tp/pull/96#discussion_r1367847714) comment) that were problematic at the UI level it became the case. While discussing with @lipwei1808 I had some suggestions to the Amount class that I added together with this PR (which is why also there are no Amount Tests yet) so that he can take over as I've informed him [prior](https://github.com/AY2324S1-CS2103-T16-3/tp/pull/97#discussion_r1368146067). I guess as opposed to finding the stash and copying over the .java file to send it manually, this seemed a bit more expedient. Either way, I can revert to the previous Amount class if this arrangement might cause any issues.